lib/block_view.go

```diff
@@ -844,6 +860,9 @@ type UtxoView struct {
 	// Coin balance entries
 	HODLerPKIDCreatorPKIDToBalanceEntry map[BalanceEntryMapKey]*BalanceEntry
 
+	// Derived Key entries. First index is owner PKID, then derived PKID
+	PKIDToDerivedKeyEntry          map[PKID]map[PKID]*DerivedKeyEntry
```

Comment:
A couple of separate thoughts here:

First, it feels like everything dealing with derived keys should be public key based. It should be PublicKey1 -> <PublicKey2, PublicKey3>. And we should leave PKID as a 1:1 mapping with each public key that's separate from all this DerivedKey stuff.

Second, this map of maps is a bit too complex. Our convention is to keep everything as <key> -> Entry. This being said, it seems like we should track everything as:
```
type DerivedKeyMapKey struct {
OwnerPublicKey []byte
DerivedPublicKey []byte
}

map[DerivedKeyMapKey]*DerivedKeyEntry
```

I know this will make the GetAllDerivedKeysForPublicKey function slower, since you will have to iterate over the whole map. But I think it's worth the simplification. I also don't see that function used anywhere.

Reply:
I'm going to pause my review here while I discuss public keys vs PKIDs for this with @maebeam.

Reply:
I changed everything to be public key based including the utxo map key. Not a huge fan that it causes `GetAllDerivedKeyMappingsForOwner` function to be `O(n)` but I agree it makes sense consistency-wise. Also, this function is used in the backend repo.